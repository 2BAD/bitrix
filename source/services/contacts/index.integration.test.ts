import { describe, expect, it } from 'vitest'
import Bitrix from './../../bitrix.js'

const WEBHOOK_URL = process.env['WEBHOOK_URL']

if (!WEBHOOK_URL) {
  throw Error('Integration tests require environmental variable `WEBHOOK_URL` to be set')
}

const { contacts } = Bitrix(WEBHOOK_URL)

describe('Contacts', () => {
  describe('fields', () => {
    it('should get all fields', async () => {
      const { result } = await contacts.fields()
      expect(result).toMatchSnapshot()
    })
  })
})
