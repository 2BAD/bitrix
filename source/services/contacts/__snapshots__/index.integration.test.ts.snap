// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts fields should get all fields 1`] = `
Object {
  "ADDRESS": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Адрес",
    "type": "string",
  },
  "ADDRESS_2": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Адрес (стр. 2)",
    "type": "string",
  },
  "ADDRESS_CITY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Город",
    "type": "string",
  },
  "ADDRESS_COUNTRY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Страна",
    "type": "string",
  },
  "ADDRESS_COUNTRY_CODE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Код страны",
    "type": "string",
  },
  "ADDRESS_POSTAL_CODE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Почтовый индекс",
    "type": "string",
  },
  "ADDRESS_PROVINCE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Область",
    "type": "string",
  },
  "ADDRESS_REGION": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Район",
    "type": "string",
  },
  "ASSIGNED_BY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Ответственный",
    "type": "user",
  },
  "BIRTHDATE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Дата рождения",
    "type": "date",
  },
  "COMMENTS": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Комментарий",
    "type": "string",
  },
  "COMPANY_ID": Object {
    "isDeprecated": true,
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Компания",
    "type": "crm_company",
  },
  "COMPANY_IDS": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "COMPANY_IDS",
    "type": "crm_company",
  },
  "CREATED_BY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Кем создан",
    "type": "user",
  },
  "DATE_CREATE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Дата создания",
    "type": "datetime",
  },
  "DATE_MODIFY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Дата изменения",
    "type": "datetime",
  },
  "EMAIL": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "E-mail",
    "type": "crm_multifield",
  },
  "EXPORT": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Участвует в экспорте контактов",
    "type": "char",
  },
  "FACE_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Привязка к лицам из модуля faceid",
    "type": "integer",
  },
  "HAS_EMAIL": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Задан e-mail",
    "type": "char",
  },
  "HAS_IMOL": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Задана открытая линия",
    "type": "char",
  },
  "HAS_PHONE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Задан телефон",
    "type": "char",
  },
  "HONORIFIC": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "statusType": "HONORIFIC",
    "title": "Обращение",
    "type": "crm_status",
  },
  "ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "ID",
    "type": "integer",
  },
  "IM": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Мессенджер",
    "type": "crm_multifield",
  },
  "LAST_NAME": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": true,
    "title": "Фамилия",
    "type": "string",
  },
  "LEAD_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Лид",
    "type": "crm_lead",
  },
  "MODIFY_BY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Кем изменен",
    "type": "user",
  },
  "NAME": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": true,
    "title": "Имя",
    "type": "string",
  },
  "OPENED": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Доступен для всех",
    "type": "char",
  },
  "ORIGINATOR_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Внешний источник",
    "type": "string",
  },
  "ORIGIN_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Идентификатор элемента во внешнем источнике",
    "type": "string",
  },
  "ORIGIN_VERSION": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Версия оригинала",
    "type": "string",
  },
  "PHONE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Телефон",
    "type": "crm_multifield",
  },
  "PHOTO": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Фотография",
    "type": "file",
  },
  "POST": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Должность",
    "type": "string",
  },
  "SECOND_NAME": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": true,
    "title": "Отчество",
    "type": "string",
  },
  "SOURCE_DESCRIPTION": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Описание",
    "type": "string",
  },
  "SOURCE_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "statusType": "SOURCE",
    "title": "Источник",
    "type": "crm_status",
  },
  "TYPE_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "statusType": "CONTACT_TYPE",
    "title": "Тип контакта",
    "type": "crm_status",
  },
  "UTM_CAMPAIGN": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Обозначение рекламной кампании",
    "type": "string",
  },
  "UTM_CONTENT": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Содержание кампании",
    "type": "string",
  },
  "UTM_MEDIUM": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Тип трафика",
    "type": "string",
  },
  "UTM_SOURCE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Рекламная система",
    "type": "string",
  },
  "UTM_TERM": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Условие поиска кампании",
    "type": "string",
  },
  "WEB": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Сайт",
    "type": "crm_multifield",
  },
}
`;
